### ISSUE TITLE
New layer names breaks deep watershed losses construction

### ISSUE THREAD

**Issue by ngreenwald** [vanvalenlab/deepcell-tf#367]
**Describe the bug**
Switching the names of the layers in the deep watershed model to include 'semantic' means that they get picked up by the loss construction logic, resulting in too many losses. 

**To Reproduce**
Run the Deep Watershed example notebook. 

**Screenshots**
![image](https://user-images.githubusercontent.com/13770365/80055070-a2a76000-84d5-11ea-9c36-461b868d06a9.png)

**Suggested fix**
We can either change the names back to not include 'semantic', or modify the following code to be more specific and not grab everything with semantic in the name
```
# Give losses for all of the semantic heads
for layer in model.layers:
    print(layer.name)
    if 'semantic_' in layer.name:
        n_classes = layer.output_shape[-1]
        loss[layer.name] = semantic_loss(n_classes)
```

**Comment by ngreenwald** [comment]
Dave sent me his fix, I'll open a PR tomorrow

**Comment by willgraf** [comment]
We could just change the conditional a little bit to use `startswith`:

```python
# Give losses for all of the semantic heads
for layer in model.layers:
    print(layer.name)
    if layer.name.startswith('semantic_'):
        n_classes = layer.output_shape[-1]
        loss[layer.name] = semantic_loss(n_classes)
```

**Comment by ngreenwald** [comment]
Dave sent me this, happy to go with whatever you guys think is best
```
# Define loss
loss_layers = ['semantic_0', 'semantic_1', 'semantic_2']
loss = {}
for layer_name in loss_layers:
    n_classes = model.get_layer(layer_name).output_shape[-1]
    if n_classes > 1:
        def loss_function(y_true, y_pred):
            return 0.01 * losses.weighted_categorical_crossentropy(
                y_true, y_pred, n_classes=n_classes)
        loss[layer_name] = loss_function
    elif n_classes == 1:
        loss[layer_name] = MSE
```
